plotly/tools.py

```diff
@@ -2275,6 +2283,62 @@ def create_candlestick(open, high, low, close,
         return dict(data=data, layout=layout)
 
 
+    @staticmethod
+    def create_dendrogram(X, orientation="bottom", labels=None,
+                          colorscale=None):
+        """
+        BETA function that returns a dendrogram Plotly figure object.
+
+        :param (ndarray) X: Matrix of observations as arrray of arrays
+        :param (str) orientation: 'top', 'right', 'bottom', or 'left'
+        :param (list) labels: List of axis category labels(observation labels)
+        :param (list) colorscale: Optional colorscale for dendrogram tree
+        clusters
+
+        X: Heatmap matrix as array of arrays
+        orientation: 'top', 'right', 'bottom', or 'left'
+        labels: List of axis category labels
+        colorscale: Optional colorscale for dendrogram tree clusters
+
+
+        Example 1: Simple bottom oriented dendrogram
+        ```
+        import plotly.plotly as py
+        from plotly.tools import FigureFactory as FF
+
+        import numpy as np
+
+        X = np.random.rand(5,5)
+        dendro_X = FF.create_dendrogram(X)
+        py.iplot(dendro_X, validate=False, height=300, width=1000)
+
+        ```
+
+        Example 2: Dendrogram to put on the left of the heatmap
+        ```
+        X = np.random.rand(5,5)
+        names_X = ['Jack', 'Oxana', 'John', 'Chelsea', 'Mark']
+        dendro_X = FF.create_dendrogram(X, orientation='right', labels=names_X)
+
+        py.iplot(dendro_X, validate=False, height=1000, width=300)
+        ```
+        """
+
+        if _scipy_imported is False:
+            raise ImportError("FigureFactory.create_dendrogram requires scipy,
+                              scipy.spatial and scipy.hierarchy")
```

Comment:
@theengineear 
from https://docs.python.org/2/library/exceptions.html
"exception ImportError
Raised when an import statement fails to find the module definition or when a from ... import fails to find a name that is to be imported." 

so I think our _scipy_imported will be false when from ... import statements fail in the beginning of the script